File: app/widgets/molecular/matrices/catalogs/action_panel.js
//= require <templates/tooltip>
//= require <widget>
//= require <molecular/matrix>


Module('Molecular.Matrices.Catalogs', function () {
  JooseClass('ActionPanel', {
    isa: Templates.ActionPanel,
    has:{
      catalog: { is: 'ro', init: function () { return this.parent(); } },
      buttons: {
        is: 'ro',
        init: [
          { label: 'Create', img: { src: "/images/small_addnew.gif" }, imode: 'edit' },
          { label: 'Delete', img: { src: "/images/small_cross.png" },  imode: 'edit' }
        ]
      }
    },
    methods: {
      onClick: function(event) {
        var me = this;
        Event.delegate({
          'input[type="button"]': function (event) {
            switch (event.element().readAttribute('value')) {
              case 'Create':
                this.viewport().widget('window').loadPage('new_project_molecular_matrix_path');
              break;
              case 'Delete':
                this.requireSelection(function(){
                  var matrix = new Molecular.MatrixView({context: this.context()});
                  matrix.deleteSelected({ collectionString: 'matrices' });
                });
              break;
            }
          }
        }).call(this, event)
      }
    }
  })
});

File: app/widgets/molecular/alignments/catalogs/action_panel.rb
module Molecular
  module Alignments
    module Catalogs
      class ActionPanel < Templates::ActionPanel
        def initialize options
          @buttons ||= [
            #{ label: 'Create', img: { src: "/images/small_addnew.gif" }, imode: 'edit' },
            { label: 'Delete', img: { src: "/images/small_cross.png" }, imode: 'edit' }
          ]
          super
        end
        #  def to_s
        #    render partial: 'molecular/alignments/catalogs/action_panel'
        #  end
      end
    end
  end
end
